### PR TITLE
test: cover daemon-backed MCP HTTP initialized lifecycle

### LINKED ISSUE #392: refactor(mcp): unify MCP session bootstrap path and add CLI coverage
## Summary

Current MCP initialization/session bootstrap is orchestrated in multiple places instead of one shared path. In practice, this has already led to protocol fixes landing in one path but not another.

The immediate example is HTTP MCP lifecycle handling around:

- `initialize`
- `notifications/initialized`
- subsequent `tools/list` / `tools/call`

Today the adapter-side HTTP path and the daemon-backed HTTP session path each coordinate this flow separately. That duplication makes it easy to fix the non-primary path while missing the main CLI/daemon path.

## Problem

UXC should not maintain multiple independent MCP bootstrap sequences.

Right now there are effectively multiple orchestration sites for MCP session readiness, especially for HTTP MCP:

- adapter-level HTTP path
- daemon-managed HTTP session creation path
- existing stdio lifecycle path with similar but separate semantics

This creates two concrete risks:

1. Protocol/lifecycle fixes can land in one path but not another.
2. Tests can pass while the default `uxc <endpoint> ...` CLI flow is still wrong, because the CLI defaults to daemon-backed execution.

## Goal

Refactor MCP lifecycle/session bootstrap so UXC keeps one canonical bootstrap path per transport/session kind, and shared callers reuse that path instead of open-coding:

1. create/connect transport
2. `initialize`
3. required post-initialize lifecycle ack(s), including `notifications/initialized` where applicable
4. return a ready-to-use session/transport

### Refactor
- Unify MCP bootstrap/session-readiness orchestration so adapter and daemon code do not each manually sequence lifecycle steps.
- Remove or minimize duplicated MCP HTTP session bootstrap logic across:
  - adapter execution/schema path
  - daemon HTTP session creation/reuse path
- Make the daemon-backed path use the same canonical readiness/bootstrap behavior as the non-daemon path.
- Review stdio and HTTP MCP bootstrap responsibilities and align the abstraction boundary around session readiness instead of individual transport calls.

### Tests
Add coverage for the real user path, not only transport/adapter internals.

Specifically add CLI/daemon-facing regression coverage that proves:

- `uxc <http-mcp-endpoint> -h` works against a server that requires `notifications/initialized`
- `uxc <http-mcp-endpoint> <tool> ...` works against the same kind of server
- daemon-backed HTTP session creation sends any required post-initialize lifecycle ack before later requests
- session reuse does not skip required lifecycle state in a way that breaks spec-compliant servers

The key requirement is that tests exercise the default CLI path, which currently routes through the daemon.

## Non-goals
- This issue is about lifecycle/bootstrap unification and test coverage.
- It is not a request to redesign all MCP transports or remove daemon-backed sessions.

## Why this matters

The current split makes regressions too easy:

- a PR can correctly fix adapter-level HTTP MCP behavior
- while the daemon-backed path used by normal CLI execution remains broken

That is a structural problem, not just a one-off bug.

### PR DESCRIPTION
**Pull request by jolestar**
## What
Add end-to-end coverage for the MCP HTTP `initialize` / `notifications/initialized` lifecycle on the default daemon-backed CLI path.

## Why
The default `uxc <endpoint> ...` flow uses the daemon, so adapter-only coverage is not enough to prevent regressions in the real user path.

Closes #392

## How
Tighten the `requires_initialized_ack` HTTP test server scenario to track initialized state per MCP session, then exercise both daemon autostart host-help and daemon-backed tool execution against that stricter lifecycle contract.

## Changes

- [x] Track `notifications/initialized` per MCP HTTP session in the local test server
- [x] Make the `requires_initialized_ack` scenario issue a session id during `initialize`
- [x] Assert daemon usage and daemon autostart in the existing local CLI lifecycle tests
- [x] Add daemon E2E coverage for autostarted host-help against a server that requires `notifications/initialized`

### Unit Tests
- [x] New regression tests added
- [ ] `cargo test` passes

### Manual Testing
- [ ] Tested against real APIs
- [x] Tested edge cases
- [x] Tested error conditions

### Test Results
```bash
cargo test --test local_e2e_test test_mcp_http_sends_initialized_ack_after_initialize -- --exact
cargo test --test local_e2e_test test_mcp_http_tool_call_works_after_initialized_ack -- --exact
cargo test --test daemon_reuse_test mcp_http_host_help_autostarts_daemon_and_sends_initialized_ack -- --exact
cargo test --test daemon_reuse_test mcp_http_daemon_session_sends_initialized_ack_before_first_request -- --exact
```

## Checklist

- [x] Code formatted with `cargo fmt`
- [ ] No clippy warnings (`cargo clippy -- -D warnings`)
- [ ] All tests pass (`cargo test`)
- [ ] Documentation updated
- [x] Commit messages follow convention
- [x] PR title follows convention

## Breaking Changes

None.

## Additional Notes

This intentionally focuses on the default daemon-backed CLI path and the lifecycle regression from the merged HTTP initialize fix. It does not try to remove the non-daemon adapter path in this PR.

### CODE REVIEW

File: src/test_server/mcp_http.rs [hunk 1]
```diff
@@ -112,31 +123,52 @@ async fn mcp_handler(
...
+            .unwrap_or(false);
+        if !initialized {
+            return Ok(Json(json!({
+                "jsonrpc": "2.0",
+                "id": req_id,
+                "error": {"code": -32002, "message": "Server not initialized"}
+            }))
+            .into_response());
+        }
     }
```
[the remark this answers is not in the document]
Comment (jolestar) [reply src/test_server/mcp_http.rs:145]: Addressed in c24daf4. The `RequiresInitializedAck` gate now uses `lock().expect(...)`, returns `400 Bad Request` when `mcp-session-id` is missing, and also returns `400` for unknown session ids instead of folding those cases into `-32002 Server not initialized`. I re-ran the relevant MCP HTTP lifecycle regressions after the change.
